### PR TITLE
Kubectl-node-shell plugin

### PR DESCRIPTION
**Pull request by kvaps**
Tiny plugin for getting shell into the node.

fixes: https://github.com/kvaps/kubectl-enter/issues/2

### CODE REVIEW

**Comment by ahmetb** [comment]
Thanks for submission.
We already have plugins like `cssh` and `ssh-jump` that sounds similar to this. Do you mind clarifying the differences for us and the users?

**Comment by kvaps** [comment]
This plugin is not uses ssh, it runs the pod and uses nsenter for getting the shell into node.

File: plugins/enter.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: enter
+spec:
+  version: "v1.0.0"
+  homepage: https://github.com/kvaps/kubectl-enter
+  shortDescription: Exec into node via kubectl 
+  description: |
+    Tiny plugin for getting shell into the node.
```
Comment (corneliusweig) [inline plugins/enter.yaml]: I think the mechanics of your plugin are simple enough to mention here. Something like

> Start a root shell in the node's host OS running. You need to be able to start privileged containers for that.

File: plugins/enter.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: enter
+spec:
+  version: "v1.0.0"
+  homepage: https://github.com/kvaps/kubectl-enter
+  shortDescription: Exec into node via kubectl
```
Comment (corneliusweig) [inline plugins/enter.yaml]:
```suggestion
  shortDescription: Spawn a root shell on a node via kubectl 
```

File: plugins/enter.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: enter
```
[the remark this answers is not in the document]
Comment (kvaps) [reply plugins/enter.yaml]: I don't like sudo, because there is no any sudo, what about `nsenter`?

[the remark this answers is not in the document]
Comment (corneliusweig) [reply plugins/enter.yaml]: :thinking: `nsenter` is not not so widely known, so you may lose many potential users here. Also, it does not make clear that you are always trying to enter the init's process namespaces. But it's better than `enter`.

What do you think about `host-shell`?

[the remark this answers is not in the document]
Comment (kvaps) [reply plugins/enter.yaml]: Kubernetes is not uses `host` term, it uses `node` instead, so I thinking `node-sh` would be fine.

[the remark this answers is not in the document]
Comment (corneliusweig) [reply plugins/enter.yaml]: Very true. We already have two `*-shell` plugins, but no `*sh` plugin. So `node-shell` would be best I think. But if you feel strongly about `node-sh` I won't argue.

File: plugins/enter.yaml [hunk 4]
```diff
@@ -5,14 +5,18 @@ metadata:
 spec:
   version: "v1.0.0"
   homepage: https://github.com/kvaps/kubectl-enter
-  shortDescription: Exec into node via kubectl 
+  shortDescription: Spawn a root shell on a node via kubectl 
   description: |
-    Tiny plugin for getting shell into the node.
+    Start a root shell in the node's host OS running. You need to be able to start privileged containers for that.
```
Comment (corneliusweig) [inline plugins/enter.yaml]: Can you wrap this at 80 characters, so it prints nicely on small terminals?

File: plugins/node-shell.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: node-shell
+spec:
+  version: "v1.0.0"
+  homepage: https://github.com/kvaps/kubectl-node-shell
+  shortDescription: Spawn a root shell on a node via kubectl 
+  description: |
+    Start a root shell in the node's host OS running.
```
Comment (corneliusweig) [inline plugins/node-shell.yaml]:
```suggestion
    Start a root shell running in the node's host OS.
```

**Comment by corneliusweig** [comment]
Alright, from my side this is good to go.
/lgtm
/approve

/cc @ahmetb Are you satisfied with the distinction to `cssh` and `ssh-jump`?

**Comment by ahmetb** [comment]
We get a lot of plugins with nearly the same functionality, but distinct in terms of the way they do that. Especially in this ssh’ing or node-related things space. 

I think we can keep accepting, as long as the description is good enough —and they don’t have massive caveats like “only works if ip is public, only works on aws” etc.

**Comment by corneliusweig** [comment]
Ok, I think this plugin is well documented and distinct enough to other solutions.

/hold cancel

**Pull request merged**